xapian-letor/feature/colltfcolllenfeature.cc

```diff
@@ -41,33 +40,35 @@ CollTfCollLenFeature::get_values() const
 {
     LOGCALL(API, std::vector<double>, "CollTfCollLenFeature::get_values", NO_ARGS);
 
-    Query query = Feature::internal->feature_query;
-    map<string, long int> coll_tf = Feature::internal->collection_termfreq();
-    map<string, long int> coll_len = Feature::internal->collection_length();
-
     vector<double> values;
     double value = 0;
 
-    for (Xapian::TermIterator qt = query.get_terms_begin(); qt != query.get_terms_end(); ++qt) {
+    for (Xapian::TermIterator qt = feature_query.get_unique_terms_begin();
+	 qt != feature_query.get_terms_end(); ++qt) {
 	if ((*qt).substr(0, 1) == "S" || (*qt).substr(1, 1) == "S")
-	    value += log10(1 + ((double)coll_len["title"] / (double)(1 + coll_tf[*qt])));
+	    value += log10(1 + ((double)collection_length.find("title")->second
+				/ (double)(1 + collection_termfreq.find(*qt)->second)));
 	else
 	    value += 0;
```

Comment:
You've not addressed this earlier feedback, despite claiming the branch was ready for merge.  There are many instances below too.